sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -434,10 +439,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
 
           logInfo(s"Listing $qualified on driver")
           Try(fs.listStatus(qualified)).getOrElse(Array.empty)
-        }.filterNot { status =>
-          val name = status.getPath.getName
-          name.toLowerCase == "_temporary" || name.startsWith(".")
-        }
+        }.filterNot(fileStatusFilter)
```

Comment:
In `ParquetRelation`, we inject cached `FileStatus`es into an anonymous extended `ParquetInputFormat`. I'd expect that disabling caching data files should cause test failures, not sure why no test failure occurred :( Either I missed something here or our test cases are rather incomplete.